### PR TITLE
Rescue master branch PRs: Template enhancements

### PR DESCRIPTION
**Pull request by GuySartorelli**
Requires
https://github.com/silverstripe/silverstripe-framework/pull/10507

Rescues
- https://github.com/silverstripe/silverstripe-framework/pull/8152
- https://github.com/silverstripe/silverstripe-framework/pull/8456
- https://github.com/silverstripe/silverstripe-framework/pull/8148

I have opted to keep these in one PR because two of them touch the template parser which needs to be recompiled whenever the .peg file is changed - it ends up being a lot easier to just do these all in one go.

## Parent issue
- https://github.com/silverstripe/silverstripe-framework/issues/10350

### CODE REVIEW

File: src/View/SSTemplateParser.php [hunk 1]
```diff
@@ -528,7 +528,7 @@ function match_CallArguments ($stack = array()) {
     			$_64 = NULL;
     			do {
     				if (( $subres = $this->whitespace(  ) ) !== FALSE) { $result["text"] .= $subres; }
-    				if (substr($this->string ?? '',$this->pos ?? 0,1) == ',') {
+    				if (substr($this->string,$this->pos,1) == ',') {
```
Comment (emteknetnz) [inline src/View/SSTemplateParser.php:531]: Removing a bunch of the null coalescing operators added by the PHP 8.1 code writing - can we guarantee that these will always not be null?  It's going to be safer to keep these in

We can run this file through https://github.com/emteknetnz/php81-fixer to put these back in

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: src/View/SSTemplateParser.php (hunk 1 above)
Comment (GuySartorelli) [reply src/View/SSTemplateParser.php:531]: Those must not have been added to the .peg file then. Those changes will have to be made on the .peg file, and then the .php file will need to be recompiled.

Comment (emteknetnz) [reply src/View/SSTemplateParser.php:531]: I've created a [PR](https://github.com/silverstripe/silverstripe-framework/pull/10507) to `4` to updated the .peg file.  We should merge this new PR, then merge it up to `5`, then rebase the PR on `5`

Comment (GuySartorelli) [reply src/View/SSTemplateParser.php:531]: I don't think the order really matters but I'm happy to do it that way.

Comment (GuySartorelli) [reply src/View/SSTemplateParser.php:531]: I've merged that PR now

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**